packages/atomic/src/components/result-template-components-v1/atomic-result-icon/atomic-result-icon.tsx

```diff
@@ -41,29 +30,20 @@ export class AtomicResultIcon {
 
     const fileType = fileTypeIcons[fileTypeValue?.toLowerCase()];
     const objectType = objectTypeIcons[objectTypeValue?.toLowerCase()];
-    return objectType || fileType || 'custom';
-  }
-
-  private get isPath() {
-    return !!this.icon?.match(/^(https?:\/\/|\.\/|\.\.\/)/);
-  }
-
-  public async componentWillLoad() {
-    const icon = this.icon || this.defaultIcon;
-    const iconPath = this.isPath ? icon : getAssetPath(`./assets/${icon}.svg`);
-    const response = await fetch(iconPath);
-    this.svg =
-      response.status === 200 || response.status === 304
-        ? await response.text()
-        : null;
+    if (!fileType && !objectType) {
+      return null;
+    }
+    return objectType || fileType;
   }
 
   public render() {
-    return (
-      <Host
-        class={!this.isPath ? this.icon || this.defaultIcon : ''}
-        innerHTML={this.svg}
-      ></Host>
+    const icon = this.icon;
+    return icon ? (
+      <atomic-icon icon={'assets://' + icon} class={icon}></atomic-icon>
+    ) : (
+      <slot>
+        <atomic-icon icon="assets://custom" class="custom"></atomic-icon>
+      </slot>
```

Comment:
If this is the conditional slot mentioned above, I'm curious how it would be used, and the role of `assets://custom`?

Alternatively, if a slot is not needed, perhaps it would be possible to reduce to:

```
const icon = this.icon || 'custom'
return <atomic-icon icon={'assets://' + icon} class={icon}></atomic-icon>
```

Reply:
`assets://custom` is a wrench icon that is displayed for results that have a filetype for which there is no corresponding icon.

The following sample displays a wrench icon when no icon for the filetype is found:
```html
<atomic-result-icon></atomic-result-icon>
```

The following sample displays a thumbnail when no icon for the filetype is found:
```html
<atomic-result-icon>
  <atomic-result-image field="thumbnail"></atomic-result-image>
</atomic-result-icon>
```